### ISSUE TITLE
It is not possible to import DNS records from a private DNS zone

### ISSUE THREAD

**Issue by dkentieso** [hashicorp/terraform-provider-azurerm#25188]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.4

### AzureRM Provider Version

3.94.0

### Affected Resource(s)/Data Source(s)

azurerm_dns_a_record

### Terraform Configuration Files

```hcl
import {
  to = azurerm_dns_a_record.myrecord
  id = "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/my-rg/providers/Microsoft.Network/privateDnsZones/myprivatedomain.com/A/myrecord"
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

The DNS record should be imported to state.

### Actual Behaviour

Planning failed. Terraform encountered an error while generating this plan.

│ Error: parsing "/subscriptions/xxx/resourceGroups/my-rg/providers/Microsoft.Network/privateDnsZones/myprivatedomain.com/A/myrecord": parsing segment "staticDnsZones": parsing the RecordType ID: the segment at position 6 didn't match

│ Expected a RecordType ID that matched:

│ > /subscriptions/xxx/resourceGroups/example-resource-group/providers/Microsoft.Network/dnsZones/dnsZoneValue/A/relativeRecordSetValue

│ The parsed Resource ID was missing a value for the segment at position 6
│ (which should be the literal value "dnsZones").

**TLDR: segment 6 has to be 'dnsZones' for the import command to run in any permutation (import block or classic command), and when it is substituted for 'dnsZones' it also fails to find the (private) record as the resource id is then wrong!  (as segment6 has to be 'privateDnsZones', but appears to be unsupported)**  

### Steps to Reproduce

terraform plan -generate-config-out=generated_resources.tf

**Comment by dkentieso** [comment]
switched to using azurerm_private_dns_a_record resource type!
